praw/objects.py

```diff
@@ -1417,27 +1439,33 @@ def __init__(self, reddit_session, subreddit_name=None, json_dict=None,
             subreddit_name = json_dict['url'].split('/')[2]
 
         info_url = reddit_session.config['subreddit_about'] % subreddit_name
+        self._case_name = subreddit_name
         super(Subreddit, self).__init__(reddit_session, json_dict, fetch,
                                         info_url)
-        self._fast_name = subreddit_name
-        self._url = reddit_session.config['subreddit'] % subreddit_name
+        self.display_name = self._case_name
+        self._url = reddit_session.config['subreddit'] % self.display_name
         # '' is the hot listing
         listings = ['new/', '', 'top/', 'controversial/', 'rising/']
-        base = (reddit_session.config['subreddit'] % self._fast_name)
+        base = (reddit_session.config['subreddit'] % self.display_name)
         self._listing_urls = [base + x + '.json' for x in listings]
 
     def __repr__(self):
         """Return a code representation of the Subreddit."""
-        return 'Subreddit(subreddit_name=\'{0}\')'.format(self._fast_name)
+        return 'Subreddit(subreddit_name=\'{0}\')'.format(self.display_name)
 
     def __unicode__(self):
         """Return a string representation of the Subreddit."""
-        return self._fast_name
+        return self.display_name
 
     def _populate(self, json_dict, fetch):
```

Comment:
I'm thinking we no longer need to overwrite this method as we can just re-set the value in the constructor. However, let's get this PR in as it's been a long work in progress. We can optimize it once it's merged.